libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.h

```diff
@@ -77,6 +77,22 @@ class ESP8266WiFiGenericClass {
         uint8_t channel(void);
 
         bool setSleepMode(WiFiSleepType_t type, uint8_t listenInterval = 0);
+        // ESP32 compatibility
+        bool setSleep(bool enable)
+        {
+            if (enable)
+            {
+                return setSleepMode(WIFI_MODEM_SLEEP);
+            }
+            else
+            {
+                return setSleepMode(WIFI_NONE_SLEEP);

```

Comment:
The default sleep mode is WIFI_LIGHT_SLEEP in esp8266. I don't know in esp32 whether there are several sleep modes. So there are three possibilities:

| choice | sleep enabled | sleep disabled |
|--|--|--|
| 1 | WIFI_MODEM_SLEEP | WIFI_NONE_SLEEP |
| 2 | WIFI_LIGHT_SLEEP | WIFI_NONE_SLEEP |
| 3 | WIFI_MODEM_SLEEP | WIFI_LIGHT_SLEEP |

You implemented number 1. I think in the general esp8266 case, number 3 is commonly used, because of discussions like in #2330 where NONE_SLEEP was proposed to those with issues. I don't remember but current consumption in this mode is quite high, so making it a common use case may not be that good.
Is there another API call in ESP32 that would allow to select between different sleep modes ?

Reply:
On ESP32:
```cpp
typedef enum {
    WIFI_PS_NONE,        /**< No power save */
    WIFI_PS_MIN_MODEM,   /**< Minimum modem power saving. In this mode, station wakes up to receive beacon every DTIM period */
    WIFI_PS_MAX_MODEM,   /**< Maximum modem power saving. In this mode, interval to receive beacons is determined by the listen_interval parameter in wifi_sta_config_t */
} wifi_ps_type_t;
```
there are two overloads:
```cpp
bool WiFiGenericClass::setSleep(bool enable);
bool WiFiGenericClass::setSleep(wifi_ps_type_t mode);
```
In the first one `enable ? WIFI_PS_MIN_MODEM : WIFI_PS_NONE`

From the docs on ESP8266 it seems that `WIFI_PS_MIN_MODEM == WIFI_LIGHT_SLEEP` and `WIFI_PS_NONE == WIFI_NONE_SLEEP`

I can copy-paste the above enum in ESP8266 and add those overrides. How about that?